2230/3/rap-battle/rap-battle.rb

```diff
@@ -0,0 +1,131 @@
+require 'trollop' # Trollop is a commandline option parser for Ruby
+require 'russian_obscenity' # Gem for filtering russian obscene language
+
+opts = Trollop.options do
+  opt %s[top-bad-words], 'Prints <i> records from the Top Bad Words', default: 0
+  # opt %s[top-words], 'Prints <i> words per people or list rappers if run without --name', :default => 30
+  # opt :name, 'Name of the rapper. Use with --top-words', :type => :string
+end
+
+# rules for ARGV parametrs
+# Trollop::educate if ARGV.empty?
+Trollop.die %s[top-bad-words], 'must be non-negative' if opts[%s[top-bad-words]] <= 0
+# p opts
+
+# Generate list of files
+# rubocop:disable Style/InverseMethods
+@common_file_list = Dir.entries('data').select { |file_name| !File.directory? file_name }
+# rubocop:enable Style/InverseMethods
+
+# Hash for data store
+@data = []
+
+# declension of words depending on the number
+# :reek:UtilityFunction
+def declension_words(num, words)
+  # rubocop:disable NestedTernaryOperator
+  words[(num = (num = num % 100) > 19 ? (num % 10) : num) == 1 ? 0 : (num > 1 && num <= 4 ? 1 : 2)]
```

Comment:
too complicated

Reply:
Is the ternary operator hard to read?

Reply:
rubocop:disable NestedTernaryOperator :)